### PR TITLE
pkg: test when PATH contians a different dune

### PR DESCRIPTION
**Pull request by gridbugs**
This behaviour would be changed by https://github.com/ocaml/dune/issues/10019

### CODE REVIEW

**Review by rgrinberg — COMMENTED** [review]
LGTM but why do you need the solving step to repro this bug? You can just write the `dune.lock` that you want directly.

**Comment by gridbugs** [comment]
Good point! I'll remove the solving step.

**Review by Leonidas-from-XIV — APPROVED** [review]
I'm positively surprised `dune` didn't try to parse the `dune` shell script as dune file.

**Comment by rgrinberg** [comment]
You might also want to include the test with `dune` being visible `PATH` in whatever processes are being spawned by the build (and install) steps

**Comment by gridbugs** [comment]
> I'm positively surprised `dune` didn't try to parse the `dune` shell script as dune file.

This only works because of `(data_only_dirs bin)`.

**Comment by gridbugs** [comment]
> You might also want to include the test with `dune` being visible `PATH` in whatever processes are being spawned by the build (and install) steps

I'm not sure what you mean by this sorry. Can you give an example.

**Comment by rgrinberg** [comment]
Just a test that does:

```
(build (run sh -c "command -v dune"))
(install (run sh -c "command -v dune"))
```

And confirms the values are correct.

**Comment by gridbugs** [comment]
Ok updated to have a second package that invokes dune as a subprocess.

File: test/blackbox-tests/test-cases/pkg/different-dune-in-path.t [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  >   cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package (name $1))
+  > EOF
+  >   cat > $1.ml <<EOF
+  > let hello = "Hello, World!"
+  > EOF
+  >   cat > dune <<EOF
+  > (library
+  >  (public_name $1))
```
[the remark this answers is not in the document]
Comment (gridbugs) [reply test/blackbox-tests/test-cases/pkg/different-dune-in-path.t]: Good call. I've removed all the compiler invocations and it's much simpler now.

**Review by rgrinberg — APPROVED** [review]

**Pull request merged**